From d08bcdc25a606da4a050b6544e82dd70f0cdc8e6 Mon Sep 17 00:00:00 2001
From: Pierre Riteau <priteau@uchicago.edu>
Date: Thu, 24 Aug 2017 18:35:41 +0100
Subject: [PATCH 1/3] Restrict Ironic node list by chassis UUID

The chassis_uuid value is read from the [ironic] section in nova.conf.
This allows each nova-compute process to export only a specific list of
Ironic nodes. With only one Ironic node per nova-compute, this allows
the use of host aggregates and Blazar reservations.
---
 nova/conf/ironic.py        | 2 ++
 nova/virt/ironic/driver.py | 1 +
 2 files changed, 3 insertions(+)

diff --git a/nova/conf/ironic.py b/nova/conf/ironic.py
index 23fdd84b36..05afda4fe8 100644
--- a/nova/conf/ironic.py
+++ b/nova/conf/ironic.py
@@ -66,6 +66,8 @@ Related options:
         min=0,
         help='Timeout (seconds) to wait for node serial console state '
              'changed. Set to 0 to disable timeout.'),
+    cfg.StrOpt('chassis_uuid',
+               help='Restrict Ironic nodes to this chassis UUID.'),
 ]
 
 
diff --git a/nova/virt/ironic/driver.py b/nova/virt/ironic/driver.py
index 562fb3f061..784674b7a5 100644
--- a/nova/virt/ironic/driver.py
+++ b/nova/virt/ironic/driver.py
@@ -479,6 +479,7 @@ class IronicDriver(virt_driver.ComputeDriver):
         """
         node_list = []
         try:
+            kwargs['chassis'] = CONF.ironic.chassis_uuid
             node_list = self.ironicclient.call("node.list", **kwargs)
         except exception.NovaException as e:
             LOG.error(_LE("Failed to get the list of nodes from the Ironic "
-- 
2.16.1

